includes/polling/applications/os-updates.inc.php

```diff
@@ -9,12 +9,18 @@
 $oid = '.1.3.6.1.4.1.8072.1.3.2.4.1.2.8.111.115.117.112.100.97.116.101.1';
 
 $rrd_name = ['app', $name, $app_id];
-$rrd_def = RrdDefinition::make()->addDataset('packages', 'GAUGE', 0);
+$rrd_def = RrdDefinition::make()
+    ->addDataset('packages', 'GAUGE', 0)
+    ->addDataset('security_updates', 'GAUGE', 0);
 
 $osupdates = snmp_get($device, $oid, $options, $mib);
+$lines = explode(',', $osupdates);
```

Comment:
There is a new format for apps, that supports versioning and uses json encoding.  Allows us to support multiple versions of the client nicely.

Reply:
I'm sorry, I don't understand your comment. Could you explain in a different way ?

Reply:
https://github.com/librenms/librenms/blob/master/includes/polling/applications/docker.inc.php#L31

Reply:
I'm sorry, I just don't understand where you're going with your sentence and how it applies to the line you highlighted. Please explain.

Reply:
This line is parsing the output of the snmp extend... we have other code for that which is much safer and can handle different clients.  The main thing is using the json_app_get() function, but also you should catch and handle errors (and implement a fallback for old extend scripts that only supply one category of updates.

Unfortunately, I couldn't find any documentation for this, so the only documentation is the code:

$ grep -lr json_app_get includes/polling/applications
includes/polling/applications/gpsd.inc.php
includes/polling/applications/mailcow-postfix.inc.php
includes/polling/applications/puppet-agent.inc.php
includes/polling/applications/ntp-client.inc.php
includes/polling/applications/mdadm.inc.php
includes/polling/applications/fail2ban.inc.php
includes/polling/applications/powermon.inc.php
includes/polling/applications/backupninja.inc.php
includes/polling/applications/certificate.inc.php
includes/polling/applications/ntp-server.inc.php
includes/polling/applications/pureftpd.inc.php
includes/polling/applications/fbsd-nfs-client.inc.php
includes/polling/applications/zfs.inc.php
includes/polling/applications/seafile.inc.php
includes/polling/applications/chronyd.inc.php
includes/polling/applications/dhcp-stats.inc.php
includes/polling/applications/redis.inc.php
includes/polling/applications/powerdns.inc.php
includes/polling/applications/docker.inc.php
includes/polling/applications/fbsd-nfs-server.inc.php
includes/polling/applications/portactivity.inc.php
includes/polling/applications/ups-apcups.inc.php